Tiltfile

```diff
@@ -1,31 +1,105 @@
+def printup(*args):
+  if config.tilt_subcommand == "up":
+    print(*args)
+
+script_dir = os.path.join(config.main_dir, 'script')
+
+default_kibana_port = 5601
+kibana_host = "localhost"
+kibana_path = ""
+
+# checks whether (dev mode) kibana is already running at given port,
+# and returns the URL it is redirected to
+def discover_kibana():
+
+    default_kibana_host = "localhost"
+
+    count = int(
+      str(
+        local("lsof -n -i :%s | grep LISTEN | wc -l | tr -d ' '" % default_kibana_port, quiet=True, echo_off=True)
+      ).strip()
+    )
+
+    if count <= 0:
+      printup("No service found on port " + str(default_kibana_port))
+      return None
+
+    is_docker_kibana = str(
+      local(
+        """curl --fail -s -w '%header{kbn-name}'""" + " http://{}:{}".format(default_kibana_host, default_kibana_port),
+        quiet=True,
+        echo_off=True
+      )
+    ).strip() == "kibana"
+
+    if (is_docker_kibana):
+      printup("Running Kibana container found on port " + str(default_kibana_port))
+      return None
+
+
+    kibana_redirect_url = str(
+      local(
+        "curl --fail -s -I -o /dev/null -w '%{redirect_url}' " + 'http://{}:{}'.format(default_kibana_host, default_kibana_port) + ' || [ $? -eq 7 ]',
+        quiet=True,
+        echo_off=True
+      )
+    ).strip().replace("http://", "")
+
+    return kibana_redirect_url
+
 custom_build(
   'apm-server',
   'DOCKER_BUILDKIT=1 docker build -t $EXPECTED_REF -f packaging/docker/Dockerfile .',
   deps = ['.'],
   ignore = ['**/*_test.go', 'tools/**', 'systemtest/**', 'docs/**'],
 )
 
+k8s_yaml(kustomize('testing/infra/k8s/overlays/local'))
+
+k8s_kind('ApmServer', image_json_path='{.spec.image}')
+k8s_kind('Elasticsearch')
+
+
+# Conditionally set up Kibana resource
+discovered_kibana_url = discover_kibana()
```

Comment:
I don't really love this as a default behaviour. It feels a bit too magical, and may lead to confusion if someone has Kibana running and didn't intend for it to be used by the Tilt setup.

What do you think about making it an explicit opt-in, via [Tilt config](https://docs.tilt.dev/tiltfile_config.html)?

Reply:
Made it opt in:
`tilt up -- --local-kibana`